File: backend/stathis/src/main/java/edu/cit/stathis/task/entity/TaskCompletion.java
package edu.cit.stathis.task.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.OffsetDateTime;

@Entity
// Unique index uq_task_completions_student_task is applied via V6 SQL after dedupe.
// Do not declare @UniqueConstraint here until that migration has been run on prod.
@Table(name = "task_completions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TaskCompletion {
    @Id
    private String physicalId;

    @Column(nullable = false)
    private String studentId;

    @Column(nullable = false)
    private String taskId;

    @Column(nullable = false)
    private boolean lessonCompleted;

    @Column(nullable = false)
    private boolean quizCompleted;

    @Column(nullable = false)
    private boolean exerciseCompleted;

    @Column(nullable = false)
    private boolean isFullyCompleted;

    @Column
    private Long totalTimeTaken;

    @Column(nullable = false)
    private OffsetDateTime startedAt;

    @Column
    private OffsetDateTime completedAt;

    @Column(name = "submitted_for_review", nullable = false)
    private boolean submittedForReview;

    @Column(name = "submitted_at")
    private OffsetDateTime submittedAt;
} 

File: mobile/stathis-mobile/app/src/main/java/citu/edu/stathis/mobile/features/tasks/data/model/TaskCompletion.kt
package citu.edu.stathis.mobile.features.tasks.data.model

// Minimal model to accept backend TaskCompletion response
data class TaskCompletion(
    val physicalId: String,
    val studentId: String,
    val taskId: String,
    val lessonCompleted: Boolean,
    val quizCompleted: Boolean,
    val exerciseCompleted: Boolean,
    val isFullyCompleted: Boolean,
    val totalTimeTaken: Long?,
    val repsPerformed: Int?,
    val startedAt: String?,
    val completedAt: String?,
    val submittedForReview: Boolean,
    val submittedAt: String?
)
